ops/model.py

```diff
@@ -1053,12 +1068,34 @@ def start(self, *service_names: str):
         """Start given service(s) by name."""
         if not service_names:
             raise TypeError('start expected at least 1 argument, got 0')
+
+        if not self.ready:
+            logger.info(f"Cannot start {service_names}. Pebble is not ready yet")
```

Comment:
Not sure about using f-strings for the logger? Perhaps `logger.info("Cannot start %s. Pebble is not ready yet", ", ".join(service_names)`?

Reply:
Additionally, I'm not sure we want to obscure this error in this fashion - this message should almost certainly be a warning/error level message, but I'm not sure this check belongs here? By embedding this check in this method in this way, we don't give much info/power to the developer?

It feels like these fundamental methods should "bubble up" exceptions to the developer, but allow them to perform some basic checking themselves using the `ready` property you've defined? @benhoyt likely has some opinions here.

Reply:
It's easy enough to reformat these so they're not using F-strings, and to use `logging.warning` or `logging.error` instead (`warning` may be more appropriate).

That said, it got added to all of these places because of the use cases raised by @johnsca in #553.

I'll change it so we bubble up a new `PebbleNotReadyError` or the (added) `UnknownServiceError`. This check is all over (and common enough that I'm tempted to make it a decorator which users can import) since trying to call any of these will get an exception from Pebble anyway, and the exception type is not always predictable.